src/lib/rest/curl_context.cpp

```diff
@@ -0,0 +1,94 @@
+#include "curl_context.h"
+
+#include <map>
+
+#include <assert.h>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+
+static pthread_mutex_t contexts_mutex = PTHREAD_MUTEX_INITIALIZER;
+static std::map<std::string, struct curl_context> contexts;
+
+struct curl_context
+get_curl_context(const std::string& key)
+{
+  struct curl_context cc = {NULL, NULL};
+  int s = pthread_mutex_lock(&contexts_mutex);
+  if(s!=0)
+  {
+       LM_X(1,("pthread_mutex_lock"));
```

Comment:
When I said 'probably' I meant 'definitely'.

A library can NEVER call exit.
NEVER EVER.

A library doesn't know who its users are. So, it can't make any decisions for them. And the decision to kill the process ... that is just the most important decision to be made.

One user may be serving perfectly 100 connections when a problematic connection comes in. Because of this 'problematic' connection the entire process dies and the 101 connections stop?
No, a library can return an error, but it must always be the user of the library (the process) that decides whether the process should die or not.